entity-view/testsuite/src/test/java/com/blazebit/persistence/view/testsuite/convert/view/ConvertEntityTest.java

```diff
@@ -0,0 +1,156 @@
+        doc = cbf.create(em, Document.class).fetch("people", "contacts.friend", "partners").getResultList().get(0);
+    }
+
+    @Test
+    public void testCloneConvert() {
+        DocumentCloneView clone = evm.convert(doc, DocumentCloneView.class);
+        PersonView ownerView = evm.convert(doc.getOwner(), PersonView.class);
+
+        assertEquals(doc.getId(), clone.getId());
+        assertEquals(doc.getAge(), clone.getAge());
+        assertEquals(doc.getName(), clone.getName());
+        assertEquals(doc.getContacts(), clone.getContacts());
+        assertEquals(ownerView, clone.getOwner());
+        assertEquals(doc.getOwner().getName(), clone.getOwner().getName());
+        assertEquals(doc.getOwner().getFriend().getName(), clone.getOwner().getFriend().getName());
+        assertEquals(doc.getPeople().get(0).getName(), clone.getPeople().get(0).getName());
+        assertEquals(doc.getPeople().get(0).getFriend().getName(), clone.getPeople().get(0).getFriend().getName());
+        assertEquals(doc.getPartners().iterator().next().getName(), clone.getPartners().iterator().next().getName());
+        assertEquals(doc.getPartners().iterator().next().getFriend().getName(), clone.getPartners().iterator().next().getFriend().getName());
+        assertTrue(doc == clone.getSource());
+    }
+
+    @Test
+    public void testCloneConvertWithBuilder() {
+        DocumentCloneView clone = evm.convertWith(doc, DocumentCloneView.class)
+                .convertAttribute("owner.friend", CreatablePersonView.class, ConvertOption.CREATE_NEW)
+                .excludeAttribute("people")
+                .convertAttribute("partners", CreatablePersonView2.class, ConvertOption.CREATE_NEW)
+                .excludeAttribute("partners.id")
+                .convert();
+        PersonView ownerView = evm.convert(doc.getOwner(), PersonView.class);
+        
+
+        assertEquals(doc.getId(), clone.getId());
+        assertEquals(doc.getAge(), clone.getAge());
+        assertEquals(doc.getName(), clone.getName());
+        assertEquals(doc.getContacts(), clone.getContacts());
+        assertEquals(ownerView, clone.getOwner());
+        assertEquals(doc.getOwner().getName(), clone.getOwner().getName());
+        assertEquals(doc.getOwner().getFriend().getName(), clone.getOwner().getFriend().getName());
+        assertNull(clone.getPartners().iterator().next().getId());
+        assertEquals(doc.getPartners().iterator().next().getName(), clone.getPartners().iterator().next().getName());
+        assertNull(clone.getPartners().iterator().next().getFriend().getId());
+        assertEquals(doc.getPartners().iterator().next().getFriend().getName(), clone.getPartners().iterator().next().getFriend().getName());
+
+        assertNull(clone.getPeople());
+        assertTrue(((EntityViewProxy) clone.getOwner().getFriend()).$$_isNew());
+        assertTrue(clone.getOwner().getFriend() instanceof CreatablePersonView);
+        assertTrue(doc == clone.getSource());
+    }
+
```

Comment:
com.blazebit.persistence.view.testsuite.convert.view.ConvertViewTest#testCloneConvert2 is a test for `.withOptionalParameter()` and `@MappingParameter`. I'm not sure if this functionality would apply to entities.